libsolidity/ast/AST.h

```diff
@@ -496,7 +499,13 @@ class ContractDefinition: public Declaration, public StructurallyDocumented, pub
 	std::vector<EnumDefinition const*> definedEnums() const { return filteredNodes<EnumDefinition>(m_subNodes); }
 	std::vector<VariableDeclaration const*> stateVariables() const { return filteredNodes<VariableDeclaration>(m_subNodes); }
 	std::vector<ModifierDefinition const*> functionModifiers() const { return filteredNodes<ModifierDefinition>(m_subNodes); }
-	std::vector<FunctionDefinition const*> definedFunctions() const { return filteredNodes<FunctionDefinition>(m_subNodes); }
+	auto definedFunctions() const {
+		return definedFunctionsByName() | ranges::views::transform([](auto const& _item) { return _item.second; });
+	}
+	auto definedFunctions(std::string const& _name) const {
+		auto&& [b, e] = definedFunctionsByName().equal_range(_name);
+		return ranges::span<decltype(*b)>(b, e) | ranges::views::transform([](auto const& _item) { return _item.second; });
+	}
```

Comment:
@ekpyron is that you

Reply:
That's what we wrote about in the chat earlier :-).
I think it has to be
```
return ranges::subrange(b, e) | ranges::views::values;
```
to work.